src/pages/Compound/useLiquidityRegistry.ts

```diff
@@ -0,0 +1,79 @@
+import { useContractKit } from '@celo-tools/use-contractkit'
+import IUniswapV2Pair from '@ubeswap/core/build/abi/IUniswapV2Pair.json'
+import { ChainId, Token } from '@ubeswap/sdk'
+import { useCallback, useEffect, useState } from 'react'
+import { AbiItem } from 'web3-utils'
+
+import { ERC20_ABI } from '../../constants/abis/erc20'
+import farmBotAbi from '../../constants/abis/FarmBot.json'
+import { useFarmRegistry } from '../Earn/useFarmRegistry'
+import { CompoundBotSummaryBase } from './useCompoundRegistry'
+
+export type LiquiditySummary = {
+  token0: Token
+  token1: Token
+  compoundBotSummary: CompoundBotSummaryBase
+  userBalance: number
+}
+
+// pools containing RFP
+const liquidityPoolAddresses = ['0x25938830FBd7619bf6CFcFDf5C37A22AB15A93cA']
+
+export const useLiquidityRegistry = () => {
+  const { address, kit } = useContractKit()
+  const [summaries, setSummaries] = useState<LiquiditySummary[]>([])
+
+  const farmSummaries = useFarmRegistry()
+
+  const call = useCallback(async () => {
+    const liquidityPoolSummaries: LiquiditySummary[] = []
+    for (const liquidityPoolAddress of liquidityPoolAddresses) {
+      const uniswapV2Pair = new kit.web3.eth.Contract(IUniswapV2Pair as AbiItem[], liquidityPoolAddress)
+
+      const token0Address = await uniswapV2Pair.methods.token0().call()
+      const token1Address = await uniswapV2Pair.methods.token1().call()
+      const userBalance = address ? await uniswapV2Pair.methods.balanceOf(address).call() : 0
+
+      const token0Contract = new kit.web3.eth.Contract(ERC20_ABI as AbiItem[], token0Address)
+      const token0Symbol = await token0Contract.methods.symbol().call()
+
+      // TODO: is token1 always the RFP token? assume it is for now
```

Comment:
it is not always!!!!

Reply:
token0 is just the first token when they are sorted alphabetically

Reply:
oh yeah haha this was my quick hack to get it working. i'll use the list of RFP addresses we have in the compoundBotRegistry (or soon to be, farmBotRegistry) to find the correct token